jdaviz/core/template_mixin.py

```diff
@@ -1312,23 +1347,26 @@ def _on_label_changed(self, msg={}):
         self.label_invalid_msg = ''
         self.label_overwrite = False
 
-    def add_results_from_plugin(self, data_item):
+    def add_results_from_plugin(self, data_item, replace=None):
         """
         Add ``data_item`` to the app's data_collection according to the default or user-provided
         label and adds to any requested viewers.
         """
         if self.label_invalid_msg:
             raise ValueError(self.label_invalid_msg)
-        data_item.meta['Plugin'] = self._plugin.__class__.__name__
-        if self.app.config == 'mosviz':
-            data_item.meta['mosviz_row'] = self.app.state.settings['mosviz_row']
 
-        replace = self.viewer.selected_reference != 'spectrum-viewer'
+        if replace is None:
+            replace = self.viewer.selected_reference != 'spectrum-viewer'
```

Comment:
replace used to be hardcoded to True for viewers other than spectrum-viewer, but in this new plugin, we definitely want to plot the trace on top of the existing image, not replacing it.